doc/ss_logger.rst

```diff
@@ -0,0 +1,144 @@
+
+Next, set `SMARTSIM_LOG_LEVEL` to `info` in the terminal::
+    export SMARTSIM_LOG_LEVEL=info
+
+When we run the script with `SMARTSIM_LOG_LEVEL=info`,
+the following output appears in stdout::
+    19:52:05 osprey.us.cray.com SmartSim[130033] INFO This is a message
+    19:52:05 osprey.us.cray.com SmartSim[130033] ERROR This is an error message
+    19:52:05 osprey.us.cray.com SmartSim[130033] WARNING This is a warning message
+
+Above, we have instructed SmartSim to print all `info` messages but ignore higher
+verbose messages (`debug`).
+
+Now set the `SMARTSIM_LOG_LEVEL` to `debug` and check the output of the program.
+Set the environment variable in the terminal like so::
+    export SMARTSIM_LOG_LEVEL=debug
+
+When we run the program once again, the following output is printed to stdout::
+    20:11:12 osprey.us.cray.com SmartSim[65385] INFO This is a message
+    20:11:12 osprey.us.cray.com SmartSim[65385] DEBUG This is a debug message
+    20:11:12 osprey.us.cray.com SmartSim[65385] ERROR This is an error message
+    20:11:12 osprey.us.cray.com SmartSim[65385] WARNING This is a warning message
+
+Notice that all log messages are visible since we set `SMARTSIM_LOG_LEVEL`
+to the highest severity log level.
+
+========================
+Example: Logging to File
+========================
+The ``log_to_file()`` function in SmartSim allows users to log messages
+to a specified file by providing a `name` to the function. The severity
+level of messages printed to the file is determined by the
+`SMARTSIM_LOG_LEVEL` variable.
+
+Begin by importing the function `get_logger` and `log_to_file`:
+
+.. code-block:: python
+
+      from smartsim.log import get_logger, log_to_file
+
+Initialize a logger for use within the Python driver script:
+
+.. code-block:: python
+
+      logger = get_logger("example_logger")
+
+Using the ``log_to_file()`` function, instruct SmartSim to create a file named
+`logger.out` to write log messages to:
+
+.. code-block:: python
+
+      log_to_file("logger.out")
+
+For the example, we add all log message severities to the script:
+
+.. code-block:: python
+
+      logger.info("This is a message")
+      logger.debug("This is a debug message")
+      logger.error("This is an error message")
+      logger.warning("This is a warning message")
+
+Set `SMARTSIM_LOG_LEVEL` to `debug` in the terminal to instruct SmartSim to print all
+log messages::
+    export SMARTSIM_LOG_LEVEL=debug
+
+When we execute the Python program,
+a file named `logger.out` is created in our working directory with the listed contents::
+    21:07:40 osprey.us.cray.com SmartSim[10950] INFO This is a message
+    21:07:40 osprey.us.cray.com SmartSim[10950] DEBUG This is a debug message
+    21:07:40 osprey.us.cray.com SmartSim[10950] ERROR This is an error message
+    21:07:40 osprey.us.cray.com SmartSim[10950] WARNING This is a warning message
+
+If the program is reran with the same file name, the file contents will be overwritten.
```

Comment:
I think this detail (append vs overwrite on secondary executions) is a good detail to move up to a details section before examples. That way, readers don't have to see every example to learn that.